dashboard/modules/state/state_head.py

```diff
@@ -339,6 +339,30 @@ async def list_runtime_envs(self, req: aiohttp.web.Request) -> aiohttp.web.Respo
         record_extra_usage_tag(TagKey.CORE_STATE_API_LIST_RUNTIME_ENVS, "1")
         return await self._handle_list_api(self._state_api.list_runtime_envs, req)
 
+    def filter_events(self, events, severity_levels, source_types, **params):
+        # If custom field parameters are provided, extract them
+        entity_name, entity_id = list(params.items())[0] if params else (None, None)
+
+        def event_filter(event):
+            # Filter 1: severity_level and source_type
+            if severity_levels and event["severity"] not in severity_levels:
+                return False
+            if source_types and event["source_type"] not in source_types:
+                return False
+
+            # Filter 2: custom_fields matching entity parameters (if provided)
+            if entity_name and entity_id:
```

Comment:
Is there any case, entity_name is None, but entity_id is not?

Reply:
Good catch, we should prevent this happening, I will fix it here

Reply:
Is it possible to happen though? Maybe we can also add assert statement

Reply:
Yes, it is possible to happen since a user could make a request with any parameters

Reply:
We don't expose the http endpoints, but the user could find the HTTP request in the chrome dev tool and could make a request with any arbitrary queries. 

For the dashboard, we will not send a request has id but no name

Reply:
Maybe we can just do `assert entity_name and entity_id`?